db-tests/src/test/java/gov/nasa/jpl/aerie/database/ExternalEventTests.java

```diff
@@ -0,0 +1,2210 @@
+      if (!ex.getSQLState().equals("23505")
+          && !ex.getMessage().contains("duplicate key value violates unique constraint \"dg_unique_valid_at\"")) {
+        throw ex;
+      }
+    }
+
+    /**
+     * A given source can have two (or more) events occurring at the same time, with all fields except the keys
+     *    (see noDuplicateEventsInSameSource). This test demonstrates that behavior without error.
+     */
+    @Test
+    void nEventsAtSameTime() throws SQLException {
+      // construct the sources and events
+      ExternalSource A = new ExternalSource(
+          "A",
+          SOURCE_TYPE,
+          DERIVATION_GROUP,
+          "2024-01-01T00:00:00Z",
+          "2024-01-01T00:00:00Z",
+          "2024-01-01T01:30:00Z",
+          CREATED_AT
+      );
+
+      ExternalEvent e1 = new ExternalEvent("a", "2024-01-01T00:00:00Z", "00:10:00", A);
+      ExternalEvent e2 = new ExternalEvent("b", "2024-01-01T00:00:00Z", "00:05:00", A);
+      ExternalEvent e3 = new ExternalEvent("c", "2024-01-01T00:00:00Z", "00:15:00", A);
+
+      // insert generic external event type, source type, and derivation group
+      insertStandardTypes();
+
+      // insert sources
+      insertExternalSource(A);
+
+      // insert events
+      insertExternalEvent(e1);
+      insertExternalEvent(e2);
+      insertExternalEvent(e3);
+
+      // all 3 keys should be present!
+      var results = getDerivedEvents();
+
+      // all ranges should only have a single element and be fully present
+      final List<DerivedEvent> expectedResults = List.of(
+          new DerivedEvent(
+              "a",
+              SOURCE_TYPE,
+              "A",
+              DERIVATION_GROUP,
+              "2024-01-01 00:00:00+00",
+              "00:10:00",
+              "{[\"2024-01-01 00:00:00+00\",\"2024-01-01 01:30:00+00\")}",
+              "2024-01-01 00:00:00+00"
+          ),
+          new DerivedEvent(
+              "b",
+              SOURCE_TYPE,
+              "A",
+              DERIVATION_GROUP,
+              "2024-01-01 00:00:00+00",
+              "00:05:00",
+              "{[\"2024-01-01 00:00:00+00\",\"2024-01-01 01:30:00+00\")}",
+              "2024-01-01 00:00:00+00"
+          ),
+          new DerivedEvent(
+              "c",
+              SOURCE_TYPE,
+              "A",
+              DERIVATION_GROUP,
+              "2024-01-01 00:00:00+00",
+              "00:15:00",
+              "{[\"2024-01-01 00:00:00+00\",\"2024-01-01 01:30:00+00\")}",
+              "2024-01-01 00:00:00+00"
+          )
+      );
+      assertEquals(expectedResults.size(), results.size());
+      assertTrue(results.containsAll(expectedResults));
+    }
+
+    /**
+     * Two events, even if totally sparse, bearing same key in same source, cannot be input. We explicitly write a test
+     *    to demonstrate that this behavior causes an error, despite being as simple as a primary key collision.
```

Comment:
Again, this is a test of Postgres's implementation of a PK and not Aerie business logic. Similar argument as to why it should be removed as `sameValid_at`